plot/plot.app.js

```diff
@@ -1157,16 +1289,128 @@
+                                                    }
+                                                    // For bar chart
+                                                
+                                                    data:{
+                                                        event:{..}
+                                                        points:{
+                                                            0:{
+                                                                data:{
+                                                                    x:[ //array of labels in bar chart xaxis],
+                                                                    y: [ //array of labels in bar chart yaxis ]
+                                                                    graphic_link_pattern: // array of links which are used to make chart clickable
+                                                                },
+                                                                fullData:{
+                                                                    // Data with default parameters used by plotly
+                                                                    alignmentgroup:"",
+                                                                    cliponxaxis: true,
+                                                                    x:[],
+                                                                    y:[]
+                                                                }
+                                                                label: //label on which click event took place
+                                                                text:""
+                                                            }
+                                                        }
+                                                    }
+                                                    
+                                                    
+                                                    */
+
+                                                    if(data.hasOwnProperty("points") && data.points[0].data.hasOwnProperty("graphic_link_pattern")){
+                                                        var idx="";
+                                                        switch(data.points[0].data.type){
+                                                            case "pie":
+                                                                idx=data.points[0].data.labels.indexOf(data.points[0].label.toString());
+                                                                break;
+                                                            case "bar":
+                                                                if(data.points[0].data.orientation=="h")
+                                                                    idx=data.points[0].data.y.indexOf(data.points[0].y.toString());
```

Comment:
I'm guessing this is similar to pie but since we have to axis of data, it will differ based on orientation. I have the same comments here about adding a comment of why you use `data.points[0].y` and `data.points[0].x` but with respect to the `data.points[0].data.y` and `data.points[0].data.x` data

Reply:
Added a comment